core/src/main/scala/org/apache/spark/scheduler/TaskResultGetter.scala

```diff
@@ -135,8 +135,9 @@ private[spark] class TaskResultGetter(sparkEnv: SparkEnv, scheduler: TaskSchedul
               logError(
                 "Could not deserialize TaskEndReason: ClassNotFound with classloader " + loader)
             case ex: Exception => // No-op
+          } finally {
```

Comment:
Usually it's a bad idea to catch all exceptions / errors, but in this case I think it would be better to change line 137 to instead catch all remaining error exceptions (so "case ex: _") and log them as an error (similar to line 135).  I think this is less confusing to the user than the default "logUncaughtExceptions" logging that will happen otherwise, and it doesn't make any difference in terms of user functionality.  If you do that too, I'd also add a comment with something like "In general we avoid catching all throwables, but in this case, the Runnable is about to complete anyway, so it's better to catch remaining errors so that we can log a sensible message to the user." 

@markhamstra does this seem reasonable?